fearless/Modules/Staking/SelectValidatorsFlow/SelectedValidatorList/SelectedValidatorListViewController.swift

```diff
@@ -0,0 +1,240 @@
+        rootView.tableView.setEditing(
+            !rootView.tableView.isEditing,
+            animated: true
+        )
+        updateEditButton()
+        updateProceedButton()
+    }
+
+    @objc private func tapProceedButton() {
+        presenter.proceed()
+    }
+}
+
+// MARK: - Localizable
+
+extension SelectedValidatorListViewController: Localizable {
+    func applyLocalization() {
+        if isViewLoaded {
+            title = R.string.localizable
+                .stakingSelectedValidatorsTitle(preferredLanguages: selectedLocale.rLanguages)
+
+            updateEditButton()
+            updateProceedButton()
+            updateHeaderView()
+        }
+    }
+}
+
+// MARK: - SelectedValidatorListViewProtocol
+
+extension SelectedValidatorListViewController: SelectedValidatorListViewProtocol {
+    func updateViewModel(_ viewModel: SelectedValidatorListViewModel) {
+        self.viewModel = viewModel
+        updateHeaderView()
+        updateProceedButton()
+    }
+
+    func reload(_ viewModel: SelectedValidatorListViewModel) {
+        self.viewModel = viewModel
+        updateProceedButton()
+        rootView.tableView.reloadData()
+    }
+
+    func didRemoveItem(at index: Int) {
+        let indexPath = IndexPath(row: index, section: 0)
+        rootView.tableView.deleteRows(at: [indexPath], with: .left)
+
+        let cellViewModels = viewModel?.cellViewModels ?? []
+        if cellViewModels.isEmpty {
+            presenter.dismiss()
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension SelectedValidatorListViewController: UITableViewDataSource {
+    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
+        viewModel?.cellViewModels.count ?? 0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let viewModel = viewModel else {
+            return UITableViewCell()
+        }
+
+        let cell = tableView.dequeueReusableCellWithType(SelectedValidatorCell.self)!
+
+        let cellViewModel = viewModel.cellViewModels[indexPath.row]
+        cell.bind(viewModel: cellViewModel)
+
+        return cell
+    }
+
+    func tableView(
+        _: UITableView,
+        commit _: UITableViewCell.EditingStyle,
+        forRowAt indexPath: IndexPath
+    ) {
+        presenter.removeItem(at: indexPath.row)
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension SelectedValidatorListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        presenter.didSelectValidator(at: indexPath.row)
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection _: Int) -> UIView? {
+        guard let viewModel = viewModel else { return nil }
+
+        let headerView: SelectedValidatorListHeaderView = tableView.dequeueReusableHeaderFooterView()
+        self.headerView = headerView
```

Comment:
I think it its unexpected behaviour. If we want to reload only the header of the section then we could get it through ```headerViewForSection:```.